src/LoginControl.test.js

```diff
@@ -6,75 +6,91 @@ const util = require("util");
 
 let loginControl;
 let loginControlInstance;
-let mockUpdateAuthenticationInfo;
-let amazonAuthorizationSpy;
+const mockUpdateAuthenticationInfo = jest.fn();
+const mockAmazonAuthorization = jest.fn();
 
 beforeEach(() => {
-  mockUpdateAuthenticationInfo = jest.fn();
+  jest.resetAllMocks();
+  Object.defineProperty(window, "amazon", {
+    value: { Login: { authorize: mockAmazonAuthorization } },
+    writable: true
+  });
   loginControl = shallow(
     <LoginControl updateAuthenticationInfo={mockUpdateAuthenticationInfo} />
   );
   loginControlInstance = loginControl.instance();
-  amazonAuthorizationSpy = jest.fn();
-
-  Object.defineProperty(window, "amazon", {
-    value: { Login: { authorize: amazonAuthorizationSpy } },
-    writable: true
-  });
 });
 
-it("renders without crashing", () => {
+it("renders LoginButton without crashing", () => {
   const wrapper = shallow(<LoginControl />);
   expect(wrapper).toMatchSnapshot();
 
   wrapper.unmount();
 });
 
-it("verifies that amazon authorization is called when login button is clicked", () => {
-  mount(<LoginControl />)
-    .find("LoginButton")
-    .find("button")
-    .simulate("click");
+it("verifies that props are passed to LoginButton Component", () => {
+  expect(Object.keys(loginControl.find("LoginButton").props()).length).toBe(1);
+  const onClickProp = loginControl.find("LoginButton").prop("onClick");
+  const handleLoginSpy = jest.spyOn(loginControlInstance, "handleLogin");
+
+  onClickProp();
+
+  expect(handleLoginSpy).toHaveBeenCalledTimes(1);
+});
+
+it("verifies that amazon authorization is called when handleLogin is called", () => {
+  loginControlInstance.handleLogin();
 
   // Verify authorize called once
-  expect(amazonAuthorizationSpy.mock.calls.length).toBe(1);
+  expect(mockAmazonAuthorization.mock.calls.length).toBe(1);
```

Comment:
Test what arguments this is being called with and then that the options contains the valid options and callback is the right function.

Reply:
Done.